src/gt4py/next/program_processors/runners/dace_fieldview/transformations/map_fusion_parallel.py

```diff
@@ -0,0 +1,168 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+"""Implements the parallel map fusing transformation.
+
+THIS FILE WAS COPIED FROM DACE TO FACILITATE DEVELOPMENT UNTIL THE PR#1625 IN
+DACE IS MERGED AND THE VERSION WAS UPGRADED.
+"""
+
+from typing import Any, Optional, Set, Union
+
+import dace
+from dace import properties, transformation
+from dace.sdfg import SDFG, SDFGState, graph, nodes
+
+from . import map_fusion_helper as mfh
+
+
+@properties.make_properties
+class MapFusionParallel(mfh.MapFusionHelper):
+    """The `MapFusionParallel` transformation allows to merge two parallel maps.
+
+    While the `MapFusionSerial` transformation fuses maps that are sequentially
+    connected through an intermediate node, this transformation is able to fuse any
+    two maps that are not sequential and in the same scope.
+
+    Args:
+        only_if_common_ancestor: Only perform fusion if both Maps share at least one
+            node as direct ancestor. This will increase the locality of the merge.
+        only_inner_maps: Only match Maps that are internal, i.e. inside another Map.
```

Comment:
They are defined in the helper class, because they apply to both the parallel and serial.
I know that if you are very strict, you should only put them at the `MapFusionHelper` however, this class is private and only exists because you can share code between the serial and parallel fusion.